pkg/asset/asset_test.go

```diff
@@ -105,3 +109,102 @@ func verifyFilesCreated(t *testing.T, dir string, expectedFiles map[string][]byt
 		t.Errorf("Expected file %q not created", f)
 	}
 }
+
+func TestLoadFileWithPattern(t *testing.T) {
+	tests := []struct {
+		name        string
+		files       map[string][]byte
+		pattern     string
+		expectFiles []*File
+		expectOK    bool
+		expectErr   bool
+	}{
+		{
+			name:        "only dirs",
+			files:       nil,
+			pattern:     "",
+			expectFiles: nil,
+			expectOK:    false,
+			expectErr:   false,
+		},
+		{
+			name:        "pattern empty",
+			files:       map[string][]byte{"foo.bar": []byte("some data")},
+			pattern:     "",
+			expectFiles: nil,
+			expectOK:    false,
+			expectErr:   false,
+		},
+		{
+			name:        "pattern doesn't match",
+			files:       map[string][]byte{"foo.bar": []byte("some data")},
+			pattern:     "bar.foo",
+			expectFiles: nil,
+			expectOK:    false,
+			expectErr:   false,
+		},
+		{
+			name:    "with contents",
+			files:   map[string][]byte{"foo.bar": []byte("some data")},
+			pattern: "foo.bar",
+			expectFiles: []*File{
+				&File{
+					Filename: "foo.bar",
+					Data:     []byte("some data"),
+				},
+			},
+			expectOK:  true,
+			expectErr: false,
+		},
+		{
+			name:    "match one file",
+			files:   map[string][]byte{"foo.bar": []byte("some data")},
+			pattern: "foo.bar",
+			expectFiles: []*File{
+				&File{
+					Filename: "foo.bar",
+					Data:     []byte("some data"),
+				},
+			},
+			expectOK:  true,
+			expectErr: false,
+		},
+		{
+			name: "match multiple files",
+			files: map[string][]byte{
+				"foo-10.bar": []byte("some data 10"),
+				"foo-0.bar":  []byte("some data 0"),
+				"foo-1.bar":  []byte("some data 1"),
+				"foo-.bar":   []byte("some data 2"),
+				"foo-b.bar":  []byte("some data 3"),
+				"foo-1.ba":   []byte("some data 4"),
+			},
+			pattern: "foo-[0-9]*.bar",
```

Comment:
This file pattern is not matching what I expect that you are thinking it matches. For example, add a file with the name `foo-10x.bar` to `files`. It will match that new file.

Reply:
Hmm, I believe the `*` applies to the `[0-9]` here, so it won't match `foo-10x.bar`, I will add a test case for it.